test/box/gh-5928-several-listen-sockets.lua

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env tarantool
+
+require('console').listen(os.getenv('ADMIN'))
+
+local function prepare_table_with_success_listen_uri(idx, default_server_addr)
+    assert(idx)
+    local table_uries = {
+        default_server_addr,
+        { string.format("%s", default_server_addr) },
+        { string.format("%sA", default_server_addr),
+          string.format("%sB", default_server_addr) },
+        { uri = string.format("%s", default_server_addr) },
+        { uri = string.format("%s", default_server_addr), transport = "plain" },
+        {
+          string.format("%sA", default_server_addr),
+          { uri = string.format("%s", default_server_addr), transport = "plain" }
+        },
+        {
+          { uri = string.format("%sA", default_server_addr), transport = "plain" },
+          { uri= string.format("%sB", default_server_addr), transport = "plain" }
+        },
+        { { uri = string.format("%s", default_server_addr), transport = {'plain', 'plain'} } },
+        { { uri = string.format("%s", default_server_addr), transport = 'plain, plain' } },
+        { { uri = string.format("%s", default_server_addr), transport = 'plain,,, plain' } },
+        default_server_addr .. "A, " .. default_server_addr .. "B",
+        default_server_addr .. "A?transport=plain, " .. default_server_addr .. "B?transport=plain",
+        default_server_addr .. "?transport=plain&transport=plain",
+        default_server_addr .. "A,,,,,, " .. default_server_addr .. "B", -- skip extra ','
+        -- skip extra '?', '&', '=' and ';'
+        default_server_addr .. "??&&transport==plain;;;plain&&transport==plain;;;plain",
+        { transport = {'plain', 'plain'}, uri = string.format("%s", default_server_addr) },
+        { { uri = string.format("%s", default_server_addr), transport = 'plain; plain' } },
+        { { uri = string.format("%s", default_server_addr), transport = 'plain;;; plain' } },
```

Comment:
You don't need to restart the server to test these URIs. `box.cfg.listen` is a dynamic parameter. You can change it without server restart. For example, `box.cfg{listen = uri1} box.cfg{listen = uri2} ...`. If you need to test the same URI in different forms, you can drop it before trying a next form. For example, `box.cfg{listen = uri1} box.cfg{listen = {}} box.cfg{listen = uri1_as_table} ...`.